### PR TITLE
Add initial support for threading

### PR DESCRIPTION
**Pull request by lumisota**

### CODE REVIEW

File: ietfdata/mailarchive.py [hunk 1]
```diff
@@ -97,7 +118,23 @@ def message_from_archive_url(self, archive_url:str) -> Message:
     def messages(self) -> Iterator[Message]:
```
Comment (csperkins) [inline ietfdata/mailarchive.py:118]: Does the return type need to change - it looks like this now returns a Tuple?

File: ietfdata/mailarchive.py [hunk 2]
```diff
@@ -51,6 +51,27 @@ def _parse_archive_url(archive_url:str) -> Tuple[str, str]:
...
+    messages: List[Tuple[int, Message]]
+    
+    def __init__(self, index: int, first_message: Message):
+        self._msg_ids =  [first_message['Message-ID']]
+        self.messages = [(index, first_message)]
+
+        
+    def should_contain(self, msg: Message):
+        return msg["In-Reply-To"] in self._msg_ids
+
```
Comment (csperkins) [inline ietfdata/mailarchive.py:65]: What about the `References` header?

File: ietfdata/mailarchive.py (hunk 1 above)
Comment (lumisota) [reply ietfdata/mailarchive.py]: Fixed.

File: ietfdata/mailarchive.py (hunk 2 above)
Comment (lumisota) [reply ietfdata/mailarchive.py:68]: Added supported for this.

Comment (csperkins) [reply ietfdata/mailarchive.py:68]: Not sure I understand this code - it looks like it only checks the first entry in `References` header, but surely needs to check all?

File: ietfdata/mailarchive.py [hunk 3]
```diff
@@ -51,6 +51,30 @@ def _parse_archive_url(archive_url:str) -> Tuple[str, str]:
...
+class MessageThread:
+    _msg_ids : List[str]
+    messages: List[Tuple[int, Message]]
+    
+    def __init__(self, index: int, first_message: Message):
+        self._msg_ids =  [first_message['Message-ID']]
+        self.messages = [(index, first_message)]
+
+        
+    def should_contain(self, msg: Message):
```
Comment (csperkins) [inline ietfdata/mailarchive.py:63]: Return type?

File: ietfdata/mailarchive.py [hunk 4]
```diff
@@ -51,6 +51,30 @@ def _parse_archive_url(archive_url:str) -> Tuple[str, str]:
...
+        if "References" in msg:
+            for msg_id in msg["References"].split():
+                return msg_id in self._msg_ids
+        return msg["In-Reply-To"] in self._msg_ids
+
+    
+    def append(self, index: int, msg: Message):
```
Comment (csperkins) [inline ietfdata/mailarchive.py:70]: Return type?

Comment (lumisota) [reply ietfdata/mailarchive.py]: Added.

File: ietfdata/mailarchive.py (hunk 3 above)
Comment (lumisota) [reply ietfdata/mailarchive.py]: Added.

File: ietfdata/mailarchive.py (hunk 2 above)
Comment (lumisota) [reply ietfdata/mailarchive.py:69]: Ah -- missing `if` statement: added.

**Pull request merged**